src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -1132,61 +1132,85 @@ int NetworkOPsImp::findTransactionsByDestination (
 // Account functions
 //
 
-bool NetworkOPsImp::getAccountObjects (Ledger::pointer lpLedger,
-    Account const& accountID, uint256 const& startAfter,
-    std::uint64_t const hint, unsigned int limit,
-    std::function <bool (SLE::ref)> func) const
+bool NetworkOPsImp::getAccountObjects (Ledger::pointer ledger,
+    Account const& account, LedgerEntryType const type, uint256 dirIndex,
+    uint256 const entryIndex, std::uint32_t const limit,
+    Json::Value& jvResult) const
 {
-    auto const rootIndex = getOwnerDirIndex (accountID);
-    auto currentIndex = rootIndex;
-    bool found = true;
+    auto const rootDirIndex = getOwnerDirIndex (account);
+    auto found = false;
 
-    if (startAfter.isNonZero ())
+    if (dirIndex.isZero ())
     {
-        found = false;
+        dirIndex = rootDirIndex;
+        found = true;
+    }
+
+    auto dir = ledger->getDirNode (dirIndex);
+    if (dir == nullptr)
+        return false;
 
-        auto const hintIndex = getDirNodeIndex (rootIndex, hint);
-        SLE::pointer hintDir = lpLedger->getSLEi (hintIndex);
-        if (hintDir != nullptr)
+    std::uint32_t i {0};
+    auto& jvObjects =
```

Comment:
Assigning and creating a new variable at the same time is a little unexpected.

Why not

    auto& jvObjects = jvResult[jss::account_objects];
    jvObjects.resize (0):